client/js/shout.js

```diff
@@ -197,6 +197,26 @@ $(function() {
 				data.msg
 			]);
 
+		chan.find(".messages .text:not(.processed)")
+			.each(function() {
+				var html = $(this).html();
+				$(this)
+					.addClass("processed")
+					// Channels names are strings (beginning with a '&' or '#' character)
+					// of length up to 200 characters.
+					// See https://tools.ietf.org/html/rfc1459#section-1.3
+					.html(html.replace(/(\W)([#&][^\x07\x2C\s]{0,199})/ig,
```

Comment:
RFC states:
> the only restriction on a
> channel name is that it may not contain any spaces (' '), a control G
> (^G or ASCII 7), or a comma (',' which is used as a list item
> separator by the protocol)